src/codegen/mod.rs

```diff
@@ -4057,7 +4057,14 @@ impl CodeGenerator for ObjCInterface {
 
         let mut impl_items = vec![];
 
+        let class_name = item.path_for_allowlisting(ctx)[1..].join("::");
+
         for method in self.methods() {
+            let name = format!("{}::{}", class_name, method.rust_name());
```

Comment:
It feels a bit awkward to check blocklisted_items here (using it on something that's not an `Item`...). Can we make Objective-C methods more similar to C++ methods?

Reply:
Unfortunately this isn't straightforward, as Objective-C methods are not turned into actual instances of `Item`, as far as I can tell. I've updated `objc_method_codegen` to be more similar to `Method::codegen_method` and added documentation reflecting the state of things. It feels like changing how Objective-C methods are more broadly treated seems out of scope for this change.